### ISSUE TITLE
Confused about decoder use last step output

### ISSUE THREAD

**Issue by CryMasK** [tensorflow/addons#652]
**System information**
Build model on Google Colab with tfa-nightly and tensorflow 2.0.

**Issue**
I tried to build an encoder-decoder model following the README page.
In the example, it needs to fit `decoder_emb_inp` into decoder, which can do something like teacher-forcing.
However, I don't have proper input for this (our model's input is time series data, and output is a label for each second)

I want the decoder can use its last step output as input, so I attempt to write some code as below.
But once I call the decoder, it must need to pass the input, even I just want to initialize the state.

I'm confused on how to use the model when testing, or something like I mentioned above.

**Code to reproduce the issue**
```
DUMMY_VALUE = -1.0
num_units = 10

data_input = Input(shape=(None, num_units))
masked_input = Masking(mask_value=DUMMY_VALUE)(data_input)
encoder_outputs, forward_state_h, forward_state_c, backward_state_h, backward_state_c = Bidirectional(LSTM(num_units, return_sequences=True, return_state=True, implementation=1))(masked_input)
state_h = Concatenate()([forward_state_h, backward_state_h])
state_c = Concatenate()([forward_state_c, backward_state_c])
encoder_state = [state_h, state_c]

# not sure it is the right way to initialize InferenceSampler
sampler = tfa.seq2seq.sampler.InferenceSampler(
    sample_fn=lambda outputs: outputs,
    sample_shape=[num_units],
    sample_dtype=tf.float32,
    end_fn=lambda sample_ids: False
)

decoder_cell = LSTMCell(num_units*2, implementation=1)
projection_layer = Dense(1, activation='sigmoid')
decoder = tfa.seq2seq.BasicDecoder(
    decoder_cell, sampler, output_layer=projection_layer)

outputs, _, _ = decoder(
    inputs=???,
    initial_state=encoder_state
    )
logits = outputs.rnn_output

model = Model(inputs=[data_input], outputs=[logits])

model.compile(optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy', K_precision, K_recall],
             sample_weight_mode='temporal')
```

-----
**More specific about my problem:**
In TF 1.x, there are different inputs between `TrainingHelper(
    inputs,
    sequence_length,
    time_major=False,
    name=None
)` and `InferenceHelper(
    sample_fn,
    sample_shape,
    sample_dtype,
    start_inputs,
    end_fn,
    next_inputs_fn=None
)`

But In TF 2.0 style, I must give `inputs` when I call the decoder.
Does it mean if I use `InferenceSampler`, I just need to pass `start_inputs` to the decoder like I use `InferenceHelper`, or I should give the decoder dummy inputs and it will auto ignore the inputs after the first one?

**Comment by seanpmorgan** [comment]
cc @guillaumekln @qlzh727 when time allows.

**Comment by guillaumekln** [comment]
When using a decoder with `tfa.seq2seq.InferenceSampler`, the `inputs` is the same as the `start_inputs` you passed to `tf.contrib.seq2seq.InferenceHelper`.

**Comment by CryMasK** [comment]
Thank you for your reply.
